app/models/grda_warehouse/enrollment_change_history.rb

```diff
@@ -32,8 +32,9 @@ def self.generate_for_date!(date: Date.current)
         perform_later(date: date.to_s)
     end
 
+    # client_ids: destination client ids
     def self.create_for_clients_on_date! client_ids:, date:
-      clients = GrdaWarehouse::Hud::Client.destination.where(id: client_ids)
+      clients = GrdaWarehouse::Hud::Client.destination.where(id: client_ids).preload(:processed_service_history)
```

Comment:
this might help a little but it looks like the job doesn't work in a way where it could easily be preloaded